src/main/java/dev/esophose/playerparticles/styles/ParticleStyle.java

```diff
@@ -20,6 +21,16 @@
      */
     List<PParticle> getParticles(ParticlePair particle, Location location);
 
+    /**
+     * Gets all the particles to display based on the style's logic
+     *
+     * @param particle The ParticlePair that contains the particle's data
+     * @param location The central location of the particles
+     * @param player The player that the particles are spawning on - null for fixed effects
+     * @return A List of PParticles to spawn
+     */
+    List<PParticle> getParticles(ParticlePair particle, Location location, Player player);
```

Comment:
What is the purpose of this? Could you give an example of when this would be used?

It should be a default interface method that should probably return null by default as to not break compatibility with plugins using the old API.

Reply:
Say you wanted to make a particle that changes color based on the player's health etc. + I saw you had a very convoluted way of finding the player in particle style celebration.